numba_dpex/core/runtime/_dpexrt_python.c

```diff
@@ -553,12 +582,16 @@ static NRT_MemInfo *DPEXRT_MemInfo_fill(NRT_MemInfo *mi,
     case 3:
     {
         uint64_t value_assign = (uint64_t)value;
-        if (is_float) {
-            double const_val = (double)value;
+        if (dest_is_float && !value_is_float) {
+            double const_val = value;
```

Comment:
Why not use a union type for double as well?

Reply:
Done.